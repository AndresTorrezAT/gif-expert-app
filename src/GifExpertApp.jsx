import { useState } from 'react';
import { AddCategory } from './components/AddCategory';

export const GifExpertApp = () => {

  const [categories, setCategories] = useState([ 'One Puch', 'Dragon Ball']);

  const onAddCategory = ( newCategory ) => {
    // console.log(newCategory);
    setCategories([ newCategory, ...categories ]);
    // setCategories( cat => [ ...cat, 'Valorant' ]);
  }

  return (
    <>
        {/* Titulo */}
        <h1>GifExpertApp</h1>

        {/* Input */}
        <AddCategory 
          // setCategories={ setCategories } 
          onNewCategory={ (value) => onAddCategory(value) }
        />

        {/* Listado de Gif */}
        <ol>
          {
            categories.map( category => {
              return <li key={ category }>{ category }</li>
            })
          }

          {/* <li>ABC</li> */}
        </ol>

            {/* Gif Item */}
    </>
  )
}

// video 6

// AyOAbVAdFoCFnl4PTCgHaZ9MGtVnxSru