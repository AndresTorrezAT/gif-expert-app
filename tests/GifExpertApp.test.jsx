import { render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";



describe('Pruebas en <GifExpertApp />', () => {

    test('el primer render del GifExpert', () => {

        const {container} = render( <GifExpertApp /> );
        expect(container).toMatchSnapshot();

        // screen.debug();
    });
});