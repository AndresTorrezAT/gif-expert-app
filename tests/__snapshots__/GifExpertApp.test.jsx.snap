// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> el primer render del GifExpert 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="from"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Puch
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;

exports[`Pruebas en <GifExpertApp /> should 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="from"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Puch
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
